# Scuole Sicure
## Il patrimonio scolastico in Italia


### Analisi tra differenti basi di dati e soggetti


## Attori coinvolti:

### OpenCUP:

http://opencup.gov.it


Il **Codice Unico Progetto (CUP)** è il “codice fiscale” di un progetto, una stringa alfanumerica di 15 caratteri che identifica univocamente un progetto di investimento pubblico finanziato con risorse pubbliche o private.

Open CUP mette a disposizione di tutti - cittadini, istituzioni ed altri enti - i dati, in formato aperto, sulle decisioni di investimento pubblico finanziate con fondi pubblici nazionali, comunitari o regionali o con risorse private registrate con il Codice Unico di Progetto. 

Importanza nei processi decisionali e di monitoraggio degli stakeholder pubblici e provati

Importanza di integrazione tra i dati di diversi stake holder istituzionali


Dati aggiornati al: 01-01-2018

Licenza: Creative Commons 4.0 Attribuzione (CC BY 4.0)

Presenti su DAF:
econ__policy.default_org_o_cup

### Italia Sicura - Cantiere scuola

L'edilizia scolastica  è tra le priorità dell'azione di Governo. Per questo è stata creata una struttura di Missione che assicura il coordinamento tra diversi Ministeri e garantisce l'impulso e il coordinamento di varie linee di finanziamento e il sostegno agli Enti locali, che operano nella gestione diretta delle scuole.

I protagonisti di questa grande opera di riqualificazione del patrimonio scolastico italiano (oltre 40.000 scuole sparse sul territorio nazionale) sono coloro che la scuola la vivono ogni giorno: insegnanti e alunni

Referente: Laura Galimberti - Coordinatrice Struttura missione scuola (PCM)

https://www.youtube.com/playlist?list=PLweUe9F1xe4uDLngGc9hwOiZBsnn062Dk

http://italiasicura.governo.it/site/home/scuole.html

Link correlati:
http://www.istruzione.it/edilizia_scolastica/anagrafe.shtml

Italiasicura per gli interventi di edilizia straordinaria
http://italiasicura.governo.it/site/home/italiasicura.html

http://italiasicura.governo.it/site/home/scuole.html

* [Cantieri Scuole](http://www.cantieriscuole.it)

* [OpenCoesione per i progetti finanziati dalle politiche di coesione](http://www.opencoesione.gov.it/)

* [OpenCantieri dedicato ai lavori di infrastruttura pubblica](http://opencantieri.mit.gov.it/)


Tutti i finanziamenti STATALI, non altri enti e fondazioni.
Il codice edificio scolastico deve essere campo.
Dal 2014 il monitoraggio è vero.
http://www.cantieriscuole.it/open_data.aspx


* [#SBLOCCASCUOLE 2016 - DPCM 27 aprile 2016](http://www.cantieriscuole.it/open_data.aspx?id=27&tipo=csv)
* [#SBLOCCASCUOLE 2017 - DM 26 aprile 2017](http://www.cantieriscuole.it/open_data.aspx?id=40&tipo=csv)
* [#Scuole Nuove - Sblocco Patto Comuni 2014-2015 - D.L. 66/2014 art. 48 c. 1](http://www.cantieriscuole.it/open_data.aspx?id=25&tipo=csv)
* [#Scuole Nuove - Sblocco Patto Province / Città Metropolitane 2015 - L. 190/2014 art. 1 c. 467](http://www.cantieriscuole.it/open_data.aspx?id=26&tipo=csv)
* [#Scuole Sicure - CIPE 30/06/2014](http://www.cantieriscuole.it/open_data.aspx?id=31&tipo=csv)
* [#Scuole Sicure - D.L. 69/2013](http://www.cantieriscuole.it/open_data.aspx?id=2&tipo=csv)
* [6000 Campanili - D.L. 69/2013 - L. 164/2014](http://www.cantieriscuole.it/open_data.aspx?id=22&tipo=csv)
* [Adeguamento Sismico - Dipartimento Protezione Civile - D.L. 269 30 settembre 2003 art. 32 bis](http://www.cantieriscuole.it/open_data.aspx?id=3&tipo=csv)
* [Adeguamento sismico – DL 269/2003 art. 32 bis – DM 943/2015](http://www.cantieriscuole.it/open_data.aspx?id=50&tipo=csv)
* [Alluvione Sardegna - Legge n.190 23/12/2014 - Art.1 comma 152](http://www.cantieriscuole.it/open_data.aspx?id=43&tipo=csv)
* [Cantieri in Comune - Delibera CIPE 38/2015](http://www.cantieriscuole.it/open_data.aspx?id=23&tipo=csv)
* [Commissario Straordinario Ricostruzione Sisma 2016 - Ordinanza n. 14 del 31 gennaio 2017](http://www.cantieriscuole.it/open_data.aspx?id=48&tipo=csv)
* [Decreto Mutui BEI prima annualità - Decreto Interministeriale 23 gennaio 2015](http://www.cantieriscuole.it/open_data.aspx?id=24&tipo=csv)
* [Decreto Mutui BEI seconda annualità - Decreto Interministeriale n. 390 del 6 giugno 2017](http://www.cantieriscuole.it/open_data.aspx?id=49&tipo=csv)
* [Fondo Kyoto - D.L. 91/2014](http://www.cantieriscuole.it/open_data.aspx?id=14&tipo=csv)
* [Messa in sicurezza elementi non strutturali - Delibere CIPE 32/2010 e 6/2012](http://www.cantieriscuole.it/open_data.aspx?id=5&tipo=csv)
* [PATTI PER IL SUD – FSC 2014-2020 – Delibera CIPE n.26 del 10/08/16](http://www.cantieriscuole.it/open_data.aspx?id=38&tipo=csv) --> ATTENZIONE, per ora escluso, errore EOF
* [Periferie - Legge n.208 del 28/12/2015](http://www.cantieriscuole.it/open_data.aspx?id=32&tipo=csv)
* [POI - Energie Rinnovabili - D.M. 24 aprile 2015](http://www.cantieriscuole.it/open_data.aspx?id=4&tipo=csv)
* [PON/POR - Fondi Strutturali 2007/2013](http://www.cantieriscuole.it/open_data.aspx?id=6&tipo=csv)
* [Quota statale 8 x 1000 IRPEF per l'edilizia scolastica - Legge 147, art. 1 comma 296](http://www.cantieriscuole.it/open_data.aspx?id=41&tipo=csv)
* [Scuole d'Abruzzo - Il Futuro in Sicurezza - Delibere CIPE 47/2009, 83/2013, 77/2015](http://www.cantieriscuole.it/open_data.aspx?id=28&tipo=csv)

### MIUR - Dataset su edifici scolastici e alunni: 

Data pubblicazione:	09/03/2017
Dati al:	31/08/2016 - 201516

http://dati.istruzione.it/opendata/opendata/catalogo/elements1/leaf/?datasetId=DS0200EDIANAGRAFESTA


http://dati.ustat.miur.it/dataset/07d1c63f-9b1a-47f0-9648-66663e59c3f6/resource/9bebad94-3c0b-469e-8e50-b474d52aeb27/download/file02_2015-16_provenienze.csv

n.b da specificare:

Informazioni anagrafiche scuole delle province autonome di Aosta, Trento e Bolzano equiparate a scuole statali.
Nel flusso di dati sono contenute le informazioni riguardanti l'anagrafe delle scuole province autonome di Aosta, Trento e Bolzano equiparate a scuole statali.

Dati 2016/2017 (i più aggiornati, non stesso anno di edilizia ma poco importa)

** Anagrafica scuole - Anno 2017-2018** (link alla cartella data, come negli altri)
   * [Informazioni anagrafiche scuole statali](http://dati.istruzione.it/opendata/opendata/catalogo/elements1/leaf/AS1718SCUANAGRAFESTAT20170731.csv)

   * [Informazioni anagrafiche scuole paritarie](http://dati.istruzione.it/opendata/opendata/catalogo/elements1/AS1718SCUANAGRAFEPAR20170731.csv)
   
   * [Informazioni anagrafiche scuole delle province autonome di Aosta, Trento e Bolzano equiparate a scuole statali](http://dati.istruzione.it/opendata/opendata/catalogo/elements1/AS1718SCUANAAUTSTAT20170731.csv)
   
   * [Informazioni anagrafiche scuole paritarie delle province autonome di Aosta, Trento e Bolzano](http://dati.istruzione.it/opendata/opendata/catalogo/elements1/AS1718SCUANAAUTPAR20170731.csv)
   
   
** Studenti **

   * [Studenti per anno di corso e fascia di eta'. Scuola statale](http://dati.istruzione.it/opendata/opendata/catalogo/elements1/leaf/ALUCORSOETASTA20170221.csv)

   * [Studenti per anno di corso e fascia di eta'. Scuola paritaria](http://dati.istruzione.it/opendata/opendata/catalogo/elements1/leaf/ALUCORSOETAPAR20170221.csv)
   

** Edilizia scuola **

   * [Elenco degli edifici scolastici statali attivi e loro locazione.](http://dati.istruzione.it/opendata/opendata/catalogo/elements1/EDIANAGRAFESTA20160831.csv)
   * [Presenza vincoli degli edifici scolastici statali (idrogeologici, sismici...)](http://dati.istruzione.it/opendata/opendata/catalogo/elements1/EDIVINCOLISTA20160831.csv)
   * [Uso di origine e data di costruzione e degli edifici scolastici statali.](http://dati.istruzione.it/opendata/opendata/catalogo/elements1/EDIETAORIGINESTA20160831.csv)
   * [Superfici e volumi degli edifici scolastici statali (mq).](http://dati.istruzione.it/opendata/opendata/catalogo/elements1/EDICONSISTENZASTA20160831.csv)
   * [Raggiungibilita' degli edifici scolastici statali (mezzi pubblici, scuolabus..)](http://dati.istruzione.it/opendata/opendata/catalogo/elements1/EDICOLLEGAMENTISTA20160831.csv)




### Registro degli Aiuti di Stato
Registro Nazionale Aiuti che integra gli aiuti e degli incentivi alle imprese
http://Registro%20nazionale%20degli%20aiuti



## Data Pipeline

In [1]:
%matplotlib inline
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
import seaborn as sns
import os
from pandas import DataFrame 
from IPython.display import display, HTML
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = "all" #setta l'esecuzione di più statement in unica cella

pd.set_option('display.max_colwidth', -1) 
pd.set_option('display.width', 5000) 
pd.set_option('display.max_columns', 100) 
# Set up formatting so larger numbers aren't displayed in scientific notation (h/t @thecapacity)
pd.set_option('display.float_format', lambda x: '%.3f' % x)

### Open Cup - Metadati

In [2]:
#metadati_cup
metadati_cup = pd.read_excel("data/Metadati_OPENDATA.xlsx", sheetname="Elenco campi  definizioni")
metadati_cup

Elenco Campi Estrazione                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          definizioni
0   CUP                                    Codice Unico di Progetto (CUP): Codice identificativo di ciascun progetto d’investimento pubblico: è definito, in genere, da uno specifico algoritmo in base al contenuto del corredo informativo inserito nel Sistema. Consiste in una stringa alfanumerica di 15 caratteri associata, in maniera biunivoca, al corredo informativo di ogni progetto. Il CUP è necessario alla realizzazione del Sistema di Monitoraggio Investimenti Pubblici.                                                                                                                                                                                                                                                                                                                                                                                                                   
1   DESCRIZIONE_SINTETICA_CUP              Campo descrittivo composto, per i progetti/interventi nell'ambito dei lavori pubblici, dai seguenti elementi separati da asterisco (*): Nome infrastruttura\ninteressata dal progetto*Indirizzo o\narea di riferimento*Descrizione intervento\n\n                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
2   ANNO_DECISIONE                         Anno in cui è presa la decisione di attuare il progetto d’investimento pubblico.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   
3   STATO_PROGETTO                         Attivo: stato che caratterizza il progetto in essere che il Sistema CUP assegna di default ad ogni progetto d’investimento pubblico. Il codice CUP, in questo caso, deve essere obbligatoriamente utilizzato in tutti i correlati documenti amministrativi e contabili.\nChiuso: stato del CUP di un progetto completato in tutte le sue parti (attività concluse e risorse erogate e in assenza di vertenze in corso); il CUP deve essere chiuso direttamente dall’utente.

### Open Cup - Dati

In [402]:
#Viene indicato che è stata usata ANSI - encoding='latin1', encoding='iso-8859-1' or encoding='cp1252'
cup = pd.read_csv("data/CUP.csv", sep='|',encoding='iso-8859-1',  error_bad_lines=False, warn_bad_lines=True)
#! 87 lines skipped 

## Carico i Cup che provengono dall'aiuto --> Tralascio, non sembra essere rilevante
#aiuti_cup = pd.read_csv("data/CUP_aiuti.csv", sep=',')
#cup_aiuti = pd.merge(cup, aiuti_cup, on="CUP" )

b'Skipping line 1907: expected 73 fields, saw 118\n'
b'Skipping line 38483: expected 73 fields, saw 138\n'
b'Skipping line 57886: expected 73 fields, saw 118\nSkipping line 62887: expected 73 fields, saw 118\n'
b'Skipping line 68501: expected 73 fields, saw 138\nSkipping line 71900: expected 73 fields, saw 138\n'
b'Skipping line 78493: expected 73 fields, saw 138\n'
b'Skipping line 87767: expected 73 fields, saw 138\n'
b'Skipping line 94426: expected 73 fields, saw 138\nSkipping line 96031: expected 73 fields, saw 119\n'
b'Skipping line 110129: expected 73 fields, saw 120\n'
b'Skipping line 116684: expected 73 fields, saw 138\nSkipping line 121471: expected 73 fields, saw 118\n'
b'Skipping line 140739: expected 73 fields, saw 137\nSkipping line 145387: expected 73 fields, saw 118\n'
b'Skipping line 147635: expected 73 fields, saw 138\nSkipping line 148836: expected 73 fields, saw 118\n'
b'Skipping line 156680: expected 73 fields, saw 118\n'
b'Skipping line 182979: expected 73 fields, s

//anaconda/envs/py36/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2717: DtypeWarning: Columns (2,4,5,6,12,17,25,26,54,55,56,61,63) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [406]:
#elenco colonne per salvare
cup_column_keep = ['CUP',
 'DESCRIZIONE_SINTETICA_CUP',
 'ANNO_DECISIONE',
 'STATO_PROGETTO',
 'COSTO_PROGETTO',
 'FINANZIAMENTO_PROGETTO',
 'CODICE_STATO',
 'STATO',
 'CODICE_AREA_GEOGRAFICA',
 'AREA_GEOGRAFICA',
 'CODICE_REGIONE',
 'REGIONE',
 'CODICE_PROVINCIA',
 'SIGLA_PROVINCIA',
 'PROVINCIA',
 'CODICE_COMUNE',
 'COMUNE',
 'SOGGETTO_TITOLARE',
 'LOCALITA_SOG_TITOLARE',
 'COMUNE_SOGGETTO_TITOLARE',
 'SOTTOCATEGORIA_SOGGETTO',
 'CATEGORIA_SOGGETTO',
 'AREA_SOGGETTO',
 'CODICE_NATURA_INTERVENTO',
 'NATURA_INTERVENTO',
 'TIPOLOGIA_INTERVENTO',
 'CODICE_AREA_INTERVENTO',
 'AREA_INTERVENTO',
 'SETTORE_INTERVENTO',
 'SOTTOSETTORE_INTERVENTO',
 'CATEGORIA_INTERVENTO',
 'SOGGETTO_RICHIEDENTE',
 'TIPOLOGIA_CUP',
 'DESCRIZIONE_INTERVENTO',
 'STRUTTURA_INFRASTRUTTURA',
 'INDIRIZZO_INTERVENTO',
 'ANNO_DELIBERA',
 'FINANZA_PROGETTO',
 'STRUMENTO_PROGRAMMAZIONE',
 'COPERTURA_FINANZIARIA'
]


ValueError: could not convert string to float: 'N'

In [428]:
#casto gli importi
cup['COSTO_PROGETTO'] = cup['COSTO_PROGETTO'].astype(float)
cup['FINANZIAMENTO_PROGETTO'] = cup['FINANZIAMENTO_PROGETTO'].astype(float)
print (cup.shape)
cup = cup[cup_column_keep]
print (cup.shape)

ValueError: could not convert string to float: 'DATO NON PRESENTE'

In [401]:
cup.head(10)

CUP                                                                                                                                                                                                                                                                                                                                                                                                                                      DESCRIZIONE_SINTETICA_CUP ANNO_DECISIONE STATO_PROGETTO COSTO_PROGETTO FINANZIAMENTO_PROGETTO CODICE_STATO   STATO CODICE_AREA_GEOGRAFICA AREA_GEOGRAFICA  CODICE_REGIONE         REGIONE CODICE_PROVINCIA SIGLA_PROVINCIA           PROVINCIA  CODICE_COMUNE                   COMUNE                                                 SOGGETTO_TITOLARE     LOCALITA_SOG_TITOLARE COMUNE_SOGGETTO_TITOLARE    SOTTOCATEGORIA_SOGGETTO                                                                 CATEGORIA_SOGGETTO           AREA_SOGGETTO  CODICE_NATURA_INTERVENTO                                          NATURA_INTERVENTO                                    TIPOLOGIA_INTERVENTO  CODICE_AREA_INTERVENTO      AREA_INTERVENTO                           SETTORE_INTERVENTO         SOTTOSETTORE_INTERVENTO                                CATEGORIA_INTERVENTO                                              SOGGETTO_RICHIEDENTE TIPOLOGIA_CUP                                                                                                                                                                                                        DESCRIZIONE_INTERVENTO                                                                                                                                                      STRUTTURA_INFRASTRUTTURA                                                                                 INDIRIZZO_INTERVENTO      ANNO_DELIBERA FINANZA_PROGETTO STRUMENTO_PROGRAMMAZIONE COPERTURA_FINANZIARIA
0  E64H15001420006  SCUOLA MATERNA VIA GARIBALDI - ISOLA DELLE FEMMINE SCUOLA MATERNA VIA SAN GIOVANNI BOSCO*VIA SAN GIOVANNI BOSCO - GARIBALDI*MANUTENZIONE STRAORDINARIA CON EFFICENTAMENTO ENERGETICO E ABBATTIMENTO BARRIERE ARCHITETTONICHE                                                                                                                                                                                                                                   2015           ATTIVO         876972         876972                 5            ITALIA  I                      ISOLE          19.000           SICILIA         82               PA              PALERMO             82043          ISOLA DELLE FEMMINE      COMUNE DI ISOLA DELLE FEMMINE - PALERMO -                         ISOLA DELLE FEMMINE (PA)  ISOLA DELLE FEMMINE      AMMINISTRAZIONI COMUNALI   ENTI TERRITORIALI E DI AMMINISTRAZIONE DEL TERRITORIO                              AMMINISTRAZIONI LOCALI 3.000                      REALIZZAZIONE DI LAVORI PUBBLICI (OPERE ED IMPIANTISTICA)  MANUTENZIONE STRAORDINARIA                             1.000                    IMMOBILI             INFRASTRUTTURE SOCIALI                       SOCIALI E SCOLASTICHE           SCUOLE MATERNE                                      COMUNE DI ISOLA DELLE FEMMINE - PALERMO -                         NORMALE       MANUTENZIONE STRAORDINARIA CON EFFICENTAMENTO ENERGETICO E ABBATTIMENTO BARRIERE ARCHITETTONICHE                                                                                                                              SCUOLA MATERNA VIA GARIBALDI - ISOLA DELLE FEMMINE SCUOLA MATERNA VIA SAN GIOVANNI BOSCO                                                                                      VIA SAN GIOVANNI BOSCO - GARIBALDI                                                                   DATO NON PRESENTE  NO               ASSENTE                  COMUNITARIA         
1  H14H16000190001  FIUME BRENTA LOCALITÀ CARBOGNA*VIA CARBOGNA*MESSA IN SICUREZZA ARGINATURA DX CON MATERIALE LAPIDEO E RISA

### ItaliaSicura - Dati

Carico tutti gli interventi fatti dallo Stato sull'edilizia scolastica (a partire dal 2004)
Verificare la data minima

In [421]:
#carico tutti i file provenienti da ItaliaSicura in un unico dataset
#cambia il nome del file ma la struttura dati e header ha lo stesso formato per tutti.
# attenzione: keep leading zeros
#https://stackoverflow.com/questions/13250046/pandas-csv-import-keep-leading-zeros-in-a-column
fileDir = os.path.dirname(os.path.realpath('__file__'))
output_folder = os.path.join(fileDir, 'data/cantieriscuole')
listing = os.listdir(output_folder)   
#loop in csv files from folder
cantieri = DataFrame()
list_ = []

for infile in listing: 
    #filename = os.path.join(fileDir, infile)
    filename = os.path.join(output_folder, infile)
    filehandle = open(filename)
    print (filehandle) 
    data = pd.read_csv(filehandle, sep="|", index_col=None, header=0, encoding='UTF-8',converters={'CODICE_EDIFICIO': lambda x: str(x)})
    list_.append(data)
cantieri = pd.concat(list_)

<_io.TextIOWrapper name='/Users/mariaclaudiabodino/DEVELOPMENT/DAF/ScuoleSicure/data/cantieriscuole/open_data_interventi_finanziamenti_edilizia_scolastica-22.csv' mode='r' encoding='UTF-8'>
<_io.TextIOWrapper name='/Users/mariaclaudiabodino/DEVELOPMENT/DAF/ScuoleSicure/data/cantieriscuole/open_data_interventi_finanziamenti_edilizia_scolastica-20.csv' mode='r' encoding='UTF-8'>
<_io.TextIOWrapper name='/Users/mariaclaudiabodino/DEVELOPMENT/DAF/ScuoleSicure/data/cantieriscuole/open_data_interventi_finanziamenti_edilizia_scolastica-21.csv' mode='r' encoding='UTF-8'>
<_io.TextIOWrapper name='/Users/mariaclaudiabodino/DEVELOPMENT/DAF/ScuoleSicure/data/cantieriscuole/open_data_interventi_finanziamenti_edilizia_scolastica-19.csv' mode='r' encoding='UTF-8'>
<_io.TextIOWrapper name='/Users/mariaclaudiabodino/DEVELOPMENT/DAF/ScuoleSicure/data/cantieriscuole/open_data_interventi_finanziamenti_edilizia_scolastica-18.csv' mode='r' encoding='UTF-8'>
<_io.TextIOWrapper name='/Users/mariaclaudiabodino

In [422]:
#rinomino il campo in CUP come su Open Cup
cantieri = cantieri.rename(columns={'Codice progetto':'CUP', "CODICE_EDIFICIO":'CODICEEDIFICIO'})


In [423]:
cantieri['Importo totale opera'] = cantieri['Importo totale opera'].str.replace(",",".").astype(float)
cantieri['Finanziamento/ Provvedimento'] = cantieri['Finanziamento/ Provvedimento'].str.replace(",",".").astype(float)
cantieri.to_csv("cantieri.csv", sep='|',index=False, header=True,  encoding='utf-8')

In [425]:
cantieri.head(10)

Linea_finanziamento   REGIONE      PROVINCIA               COMUNE COD_COMUNE                     FONTE Data_aggiornamento_lavori Data_inizio   Data_fine       ENTE_RICHIEDENTE TIPOLOGIA_ENTE_RICHIEDENTE CODICEEDIFICIO   COD_PRINC     COD_MPI                 NOME_SCUOLA OGGETTO_INTERVENTO INFO_AGGIUNTIVE CENTRO_COMUNE  LATITUDINE  LONGITUDINE                                                                                                                                                                                                    TIPO_INTERVENTO     STATO LAVORI              CUP  Importo totale opera  Finanziamento/ Provvedimento
0  Quota statale 8 x 1000 IRPEF per l'edilizia scolastica - Legge 147, art. 1 comma 296  Veneto    Vicenza        Rosà                 24087      Amministrazione Comunale  2017-10-18                2018-06-10  2018-12-01  Comune di Rosà         COMUNE                     0240871554     VIIC85700   VIEE857022  ROSA` - CAPOLUOGO -         NaN                NaN             No           45.727      11.760        Miglioramento sismico ed efficientamento energetico                                                                                                                                                                Cantiere aperto  D15I16000630005 2250000.000           1221886.000                  
1  Quota statale 8 x 1000 IRPEF per l'edilizia scolastica - Legge 147, art. 1 comma 296  Marche    Ascoli Piceno  Ripatransone         44063      Amministrazione Comunale  2017-09-01                2018-06-10  2018-12-31  Comune di Ripatrasone  COMUNE                     0440630091     APIC804003  APMM804025  RIPATRANSONE "CONSORTI" MM  NaN                NaN             No           43.001      13.764        Adeguamento sismico, adeguamento igienico sanitario                                                                                                                                                                Cantiere aperto  G51E15000490002 800000.000            723949.080                   
2  Quota statale 8 x 1000 IRPEF per l'edilizia scolastica - Legge 147, art. 1 comma 296  Abruzzo   Teramo         Colledara            67018      Amministrazione Comunale  2017-09-01                2018-06-10  2018-12-01  Comune di Colledara    COMUNE                     0670180093     TEIC81900X  TEEE819045  NaN                         NaN                NaN             Si           42.540      13.676        Messa in sicurezza, adeguamento energetico, rimozione amianto                                                                                                                                                      Cantiere aperto  C86G17000100001 428654.060            428654.060                   
0  PON/POR - Fondi Strutturali 2007/2013                                                 Campania  Caserta        Vairano Patenora     061095     MIUR                      2016-04-05                NaN         NaN         Vairano Patenora       COMUNE                                    NaN         NaN         NaN                         NaN                POR             Si           41.333      14.129        risparmio energetico garantire la sicurezza degli edifici scolastici -messa a norma degli impianti aumentare l`attrattivit� degli istituti scolastici garantire l`accessibilit� a tutti negli istituti scolastici  Concluso         NaN             0.000                 349754.470                   
1  PON/POR - Fondi Strutturali 2007/2013                                                 Campania  Avellino       Vallata              064114     MIUR                      2016-04-05                NaN         NaN         Vallata                PROVINCIA                                 NaN         NaN         NaN                         NaN                POR             Si           41.033      15.253        risparmio energetico garantire l`accessibilit� a tutti negli istituti scolastici                              

In [426]:
#salva solo i cantieri di Cuneo
cantieri_cn=cantieri.loc[cantieri["PROVINCIA"]=="Cuneo"]
cantieri_cn.shape
cantieri_cn.to_csv("cantieri_cn.csv", sep='|',index=False, header=True,  encoding='utf-8')

#record non riconducibili al codice edificio
print("Codice EDIFICIO null su Cuneo: ",len(cantieri_cn[cantieri_cn["CODICEEDIFICIO"].isnull()]))
print("Codice progetto/CUP null Cuneo: ",len(cantieri_cn[cantieri_cn["CUP"].isnull()]))

(234, 25)

Codice EDIFICIO null su Cuneo:  0
Codice progetto/CUP null Cuneo:  13


### MIUR - Dati - Alunni per edifici scolastici

NB: i nomi colonne hanno uno space iniziale - esegui strip()

In [427]:
# Creo il dataframe SCUOLE con le info basi delle scuole
#Carico file di anagrafica base delle scuole
fileDir = os.path.dirname(os.path.realpath('__file__'))
output_folder = os.path.join(fileDir, 'data/MIUR/anagrafica')
listing = os.listdir(output_folder)   
#loop in csv files from folder
scuole = DataFrame()
list_ = []

for infile in listing: 
    #filename = os.path.join(fileDir, infile)
    filename = os.path.join(output_folder, infile)
    filehandle = open(filename)
    print (filehandle) 
    data = pd.read_csv(filehandle, sep=",", index_col=None, header=0)
    list_.append(data)
scuole = pd.concat(list_)

scuole = scuole.rename(columns=lambda x: x.strip())

<_io.TextIOWrapper name='/Users/mariaclaudiabodino/DEVELOPMENT/DAF/ScuoleSicure/data/MIUR/anagrafica/AS1718SCUANAAUTSTAT20170731.csv' mode='r' encoding='UTF-8'>
<_io.TextIOWrapper name='/Users/mariaclaudiabodino/DEVELOPMENT/DAF/ScuoleSicure/data/MIUR/anagrafica/AS1718SCUANAGRAFESTAT20170731.csv' mode='r' encoding='UTF-8'>
<_io.TextIOWrapper name='/Users/mariaclaudiabodino/DEVELOPMENT/DAF/ScuoleSicure/data/MIUR/anagrafica/AS1718SCUANAGRAFEPAR20170731.csv' mode='r' encoding='UTF-8'>
<_io.TextIOWrapper name='/Users/mariaclaudiabodino/DEVELOPMENT/DAF/ScuoleSicure/data/MIUR/anagrafica/AS1718SCUANAAUTPAR20170731.csv' mode='r' encoding='UTF-8'>


In [356]:
scuole.shape
#scuole_columns = list(scuole)
scuole_columns_to_keep =[
 'AREAGEOGRAFICA',
 'CAPSCUOLA',
 'CODICECOMUNESCUOLA',
 'CODICEISTITUTORIFERIMENTO',
 'CODICESCUOLA',
 'DENOMINAZIONEISTITUTORIFERIMENTO',
 'DENOMINAZIONESCUOLA',
 'DESCRIZIONECOMUNE',
 'DESCRIZIONETIPOLOGIAGRADOISTRUZIONESCUOLA',
 'INDIRIZZOSCUOLA',
 'PROVINCIA',
 'REGIONE']
#elimino le colonne non necessarie
scuole = scuole[scuole_columns_to_keep]
scuole.shape

(65503, 19)

(65503, 12)

In [357]:
scuole_cn=scuole.loc[scuole["PROVINCIA"]=="CUNEO"]
print ("Tot codici scuola univoca per provincia CUNEO: ",scuole_cn.CODICESCUOLA.nunique())
scuole_cn.groupby(["DESCRIZIONETIPOLOGIAGRADOISTRUZIONESCUOLA"]).count()


Tot codici scuola univoca per provincia CUNEO:  786


AREAGEOGRAFICA  CAPSCUOLA  CODICECOMUNESCUOLA  CODICEISTITUTORIFERIMENTO  CODICESCUOLA  DENOMINAZIONEISTITUTORIFERIMENTO  DENOMINAZIONESCUOLA  DESCRIZIONECOMUNE  INDIRIZZOSCUOLA  PROVINCIA  REGIONE
DESCRIZIONETIPOLOGIAGRADOISTRUZIONESCUOLA                                                                                                                                                                                                               
CENTRO TERRITORIALE                                4               4          4                   4                          4             4                                 4                    4                  4                4          4      
IST  PROF  PER I SERVIZI COMMERCIALI               1               1          1                   1                          1             1                                 1                    1                  1                1          1      
IST PROF INDUSTRIA E ARTIGIANATO                   8               8          8                   8                          8             8                                 8                    8                  8                8          8      
IST PROF PER I SERVIZI ALBERGHIERI E RISTORAZIONE  6               6          6                   6                          6             6                                 6                    6                  6                6          6      
IST PROF PER I SERVIZI COMMERCIALI E TURISTICI     10              10         10                  10                         10            10                                10                   10                 10               10         10     
IST PROF PER L'AGRICOLTURA E L'AMBIENTE            4               4          4                   4                          4             4                                 4                    4                  4                4          4      
IST TEC COMMERCIALE E PER GEOMETRI                 6               6          6                   6                          6             6                                 6                    6                  6                6          6      
ISTITUTO COMPRENSIVO                               59              59         59                  59                         59            59                                59                   59                 59               59         59     
ISTITUTO D'ARTE                                    2               2          2                   2                          2             2                                 2                    2                  2                2          2      
ISTITUTO MAGISTRALE                                4               4          4                   4                          4             4                                 4                    4                  4                4          4      
ISTITUTO SUPERIORE                                 16              16         16                  16                         16            16                                16                   16                 16               16         16     
ISTITUTO TECNICO AGRARIO                           4               4          4                   4                          4             4                                 4                    4                  4                4          4      
ISTITUTO TECNICO COMMERCIALE                       8               8          8                   8                          8             8                                 8                    8                  8                8          8      
ISTITUTO TECNICO INDUSTRIALE                       11              11         11                  11                         11            11                                11                   11                 11               11         11     
ISTITUTO TECNICO PER GEOMETRI                      5               

In [254]:
#ISTITUTI STATALI - raggruppo per CODICESCUOLA
alunni_statali = pd.read_csv("data/MIUR/alunni/ALUCORSOETASTA20170221.csv", sep=',')

#clean columns name
alunni_statali = alunni_statali.rename(columns=lambda x: x.strip())
print ("Unique edifici statali: ", alunni_statali["CODICESCUOLA"].nunique())
alunni_statali = alunni_statali[["CODICESCUOLA", "ALUNNI"]]
alunni_statali= alunni_statali.groupby(['CODICESCUOLA']).sum()
alunni_statali["Tipo"]='S'
alunni_statali.reset_index()

Unique edifici statali:  28272


CODICESCUOLA  ALUNNI Tipo
0      AGEE01102E   263     S  
1      AGEE01104L   364     S  
2      AGEE02405Q   106     S  
3      AGEE02406R   198     S  
4      AGEE02407T   189     S  
5      AGEE02408V   53      S  
6      AGEE034016   197     S  
7      AGEE03406B   245     S  
8      AGEE03407C   250     S  
9      AGEE035012   339     S  
10     AGEE805015   31      S  
11     AGEE805026   110     S  
12     AGEE80702T   151     S  
13     AGEE80703V   98      S  
14     AGEE80801L   320     S  
15     AGEE80802N   145     S  
16     AGEE81001L   275     S  
17     AGEE81002N   18      S  
18     AGEE812018   96      S  
19     AGEE812029   106     S  
20     AGEE81203A   166     S  
21     AGEE813014   156     S  
22     AGEE81501Q   130     S  
23     AGEE81502R   117     S  
24     AGEE81701B   272     S  
25     AGEE818017   175     S  
26     AGEE818028   168     S  
27     AGEE819013   179     S  
28     AGEE819024   219     S  
29     AGEE820017   88      S  
...           ...   ..     ..  
28242  VVMM83505R   23      S  
28243  VVMM83506T   26      S  
28244  VVMM83601C   115     S  
28245  VVPC00201Q   124     S  
28246  VVPC00701V   420     S  
28247  VVPC04000D   206     S  
28248  VVPM01000T   1217    S  
28249  VVPM030003   131     S  
28250  VVPS00201V   252     S  
28251  VVPS00301P   186     S  
28252  VVPS01000R   1148    S  
28253  VVPS07000C   86      S  
28254  VVRC00201B   269     S  
28255  VVRC009016   406     S  
28256  VVRC00951G   70      S  
28257  VVRH00301C   233     S  
28258  VVRH00302D   56      S  
28259  VVRH01000E   633     S  
28260  VVRH01001G   56      S  
28261  VVRH01050X   83      S  
28262  VVRI00901V   327     S  
28263  VVRI02000L   95      S  
28264  VVSD00701C   285     S  
28265  VVTD00301E   280     S  
28266  VVTD01000L   462     S  
28267  VVTD01004R   74      S  
28268  VVTF01101Q   403     S  
28269  VVTF04000P   210     S  
28270  VVTH01000A   354     S  
28271  VVTL01101X   156     S  

[28272 rows x 3 columns]

In [20]:
#ISTITUTI PARITARI - raggruppo per CODICESCUOLA
alunni_paritari = pd.read_csv("data/MIUR/alunni/ALUCORSOETAPAR20170221.csv", sep=',')
#clean columns name
alunni_paritari = alunni_paritari.rename(columns=lambda x: x.strip())
alunni_paritari = alunni_paritari[["CODICESCUOLA", "ALUNNI"]]
print ("Unique edifici paritari: ", alunni_paritari["CODICESCUOLA"].nunique())
alunni_paritari= alunni_paritari.groupby(['CODICESCUOLA']).sum()
alunni_paritari["Tipo"]='P'
alunni_paritari.reset_index()

Unique edifici paritari:  3587


CODICESCUOLA  ALUNNI Tipo
0     AG1E00500B   67      P  
1     AG1E006007   116     P  
2     AG1EVA500E   13      P  
3     AGPC4A5009   12      P  
4     AGPM00500N   30      P  
5     AGPS02500T   19      P  
6     AGPSCE5001   26      P  
7     AGPSFG5009   29      P  
8     AGPSI1500H   38      P  
9     AGPSTP5000   43      P  
10    AGSL015004   38      P  
11    AGTAN5500G   38      P  
12    AGTD00500C   77      P  
13    AGTD02500N   33      P  
14    AGTD105009   21      P  
15    AGTD12500E   36      P  
16    AGTD2M5007   33      P  
17    AGTD4U500B   21      P  
18    AGTD685001   42      P  
19    AGTD96500X   73      P  
20    AGTDE45003   65      P  
21    AGTL00500V   18      P  
22    AGTL4P500L   27      P  
23    AL1E00100P   287     P  
24    AL1E00300A   80      P  
25    AL1E004006   84      P  
26    AL1E005002   149     P  
27    AL1E00600T   127     P  
28    AL1E6U500H   28      P  
29    AL1M00400B   40      P  
...          ...   ..     ..  
3557  VRTD025005   75      P  
3558  VRTD04500A   100     P  
3559  VRTF005006   800     P  
3560  VRTF035002   21      P  
3561  VRTN01500C   47      P  
3562  VRTN025003   111     P  
3563  VT1E001006   109     P  
3564  VT1E002002   122     P  
3565  VT1E00300T   87      P  
3566  VT1E00400N   63      P  
3567  VT1E00500D   132     P  
3568  VT1E006009   84      P  
3569  VT1M003003   34      P  
3570  VT1M00400V   51      P  
3571  VT1M00500P   31      P  
3572  VT1M00600E   55      P  
3573  VTPL00500D   8       P  
3574  VTPLR35005   10      P  
3575  VTPM01500A   14      P  
3576  VTPM57500A   17      P  
3577  VTPM7D500R   12      P  
3578  VTPMI8500B   71      P  
3579  VTPS00500P   31      P  
3580  VTPS015009   54      P  
3581  VTTD015005   86      P  
3582  VTTD1P500U   35      P  
3583  VTTE005002   15      P  
3584  VV1E00100R   84      P  
3585  VVPS2U500I   67      P  
3586  VVSD00500Q   57      P  

[3587 rows x 3 columns]

In [21]:
#uniamo i due dataset su CODICESCUOLA e numero alunni in uno unico
alunni= pd.concat([alunni_statali, alunni_paritari])
alunni = alunni.reset_index()

In [24]:
# join scuole e alunni
print ("Iniziale: ", len(scuole_cn), "  ", len(alunni))
scuole_alunni_cn = pd.merge(scuole_cn, alunni, how="left",on='CODICESCUOLA')
print ("Finale: ",len(scuole_alunni_cn))

Iniziale:  786    31859
Finale:  786


In [27]:
scuole_alunni_cn
scuole_alunni_cn.to_csv("scuoleAlunni_cn.csv", sep='|',index=False, header=True,  encoding='utf-8')

AREAGEOGRAFICA        CAPSCUOLA CODICECOMUNESCUOLA CODICEISTITUTORIFERIMENTO CODICESCUOLA DENOMINAZIONEISTITUTORIFERIMENTO                                          DENOMINAZIONESCUOLA DESCRIZIONECARATTERISTICASCUOLA   DESCRIZIONECOMUNE DESCRIZIONETIPOLOGIAGRADOISTRUZIONESCUOLA INDICAZIONESEDEDIRETTIVO INDICAZIONESEDEOMNICOMPRENSIVO               INDIRIZZOEMAILSCUOLA INDIRIZZOPECSCUOLA                 INDIRIZZOSCUOLA PROVINCIA   REGIONE                           SITOWEBSCUOLA  ANNOSCOLASTICO  ALUNNI Tipo
0    NORD OVEST     Non Disponibile  H727               CNIS014001                CNTD014028   SALUZZO - "C. DENINA"            CASA CIRCONDARIALE MORANDI                                   SPEC. PER CARCERARI             SALUZZO             IST TEC COMMERCIALE E PER GEOMETRI        NO                       Non Disponibile                Non Disponibile                    Non Disponibile    REGIONE BRONDA 19/B             CUNEO     PIEMONTE  http://www.denina.it                    201718         nan      NaN
1    NORD OVEST     12080            G561               CNIC85900A                CNMM85902C   MONDOVI' 2                       MONDOVI' CORDERO SS PIANFEI                                  NORMALE                         PIANFEI             SCUOLA PRIMO GRADO                        NO                       Non Disponibile                Non Disponibile                    Non Disponibile    VIA ROMA 99/F                   CUNEO     PIEMONTE  www.comprensivocuneocorsosoleri.gov.it  201718         nan      NaN
2    NORD OVEST     12037            H727               CNIC862006                CNAA862013   SALUZZO                          SALUZZO-ILARIA ALPI                                          NORMALE                         SALUZZO             SCUOLA INFANZIA                           NO                       Non Disponibile                Non Disponibile                    Non Disponibile    VIA BARGE, 6/A                  CUNEO     PIEMONTE  www.comprensivocuneoviasobrero.gov.it   201718         nan      NaN
3    NORD OVEST     12037            H727               CNIC862006                CNAA862024   SALUZZO                          SALUZZO-VIA MARIA LUISA ALESSI                               NORMALE                         SALUZZO             SCUOLA INFANZIA                           NO                       Non Disponibile                Non Disponibile                    Non Disponibile    VIA MARIA LUISA ALESSI          CUNEO     PIEMONTE  www.comprensivocuneoviasobrero.gov.it   201718         nan      NaN
4    NORD OVEST     12030            H727               CNIC862006                CNAA862035   SALUZZO                          SALUZZO-FRAZ.CERVIGNASCO                                     NORMALE                         SALUZZO             SCUOLA INFANZIA                           NO                       Non Disponibile                Non Disponibile                    Non Disponibile    VIA REVALANCA, 14               CUNEO     PIEMONTE  www.comprensivocuneoviasobrero.gov.it   201718         nan      NaN
5    NORD OVEST     12030            G240               CNIC862006                CNAA862046   SALUZZO                          PAGNO                                                        NORMALE                         PAGNO               SCUOLA INFANZIA                           NO                       Non Disponibile                Non Disponibile                    Non Disponibile    VIA CADUTI LIBERAZIONE, 8       CUNEO     PIEMONTE  www.comprensivocuneoviasobrero.gov.it   201718         nan      NaN
6    NORD OVEST     12042            B111               CNIC863002                CNAA86301V   BRA 2                            BRA-VIA OSP. " G. RODARI"                                    NORMALE                         BRA                 SCUOLA INFANZIA                           NO                       Non Disponibile                Non Disponibile                    Non Disponibile    VIA OSPEDALE, 10    

### Edilizia scuola

In [366]:
#Localizzazione  degli edifici scolastici statali attivi - una scuola può avere più edifici
#NB il codiceedificio va forzato e letto con uno space 
edifici = pd.read_csv("data/MIUR/edilizia/EDIANAGRAFESTA20160831.csv", sep=',',converters={' CODICEEDIFICIO': lambda x: str(x)})
edifici = edifici.rename(columns=lambda x: x.strip())
edifici.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50804 entries, 0 to 50803
Data columns (total 11 columns):
ANNOSCOLASTICO            50804 non-null int64
CODICESCUOLA              50804 non-null object
CODICEEDIFICIO            50804 non-null object
CODICECOMUNE              50804 non-null object
DESCRIZIONECOMUNE         50804 non-null object
SIGLAPROVINCIA            49503 non-null object
TIPOLOGIAINDIRIZZO        50804 non-null object
DENOMINAZIONEINDIRIZZO    50804 non-null object
NUMEROCIVICO              50804 non-null object
CAP                       50804 non-null object
STATOEDIFICIO             50804 non-null object
dtypes: int64(1), object(10)
memory usage: 4.3+ MB


In [367]:
#mantengo solo le info che mi permettono di fare join
#edifici_columns = list(edifici)
edifici_columns_to_keep = [
 'CODICESCUOLA',
 'CODICEEDIFICIO',
 'SIGLAPROVINCIA',  
 'STATOEDIFICIO']
edifici = edifici[edifici_columns_to_keep]
edifici.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50804 entries, 0 to 50803
Data columns (total 4 columns):
CODICESCUOLA      50804 non-null object
CODICEEDIFICIO    50804 non-null object
SIGLAPROVINCIA    49503 non-null object
STATOEDIFICIO     50804 non-null object
dtypes: object(4)
memory usage: 1.6+ MB


In [368]:
#isolo gli edifici di CUNEO
edifici_cn = edifici.loc[edifici["SIGLAPROVINCIA"]=="CN"]

In [369]:
edifici_cn.shape

(698, 4)

In [370]:
#VINCOLI degli edifici scolastici statali 
edifici_v = pd.read_csv("data/MIUR/edilizia/EDIVINCOLISTA20160831.csv", sep=',',converters={' CODICEEDIFICIO': lambda x: str(x)})
edifici_v = edifici_v.rename(columns=lambda x: x.strip())
edifici_v.info()
edifici_v["VINCOLIZONASISMICA"].value_counts()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50804 entries, 0 to 50803
Data columns (total 8 columns):
ANNOSCOLASTICO              50804 non-null int64
CODICESCUOLA                50804 non-null object
CODICEEDIFICIO              50804 non-null object
VINCOLIIDROGEOLOGICI        50804 non-null object
VINCOLIPAESAGGIO            50804 non-null object
EDIFICIOVETUSTO             50804 non-null object
VINCOLIZONASISMICA          50804 non-null object
PROGETTAZIONEANTISISMICA    50804 non-null object
dtypes: int64(1), object(7)
memory usage: 3.1+ MB


Rischio sismicità medio/alta (S=9)    18622
Rischio bassa sismicità (S=6)         14970
Non classificato a rischio            13018
Rischio elevata sismicità (S=12)      3578 
Non Comunicato                        616  
Name: VINCOLIZONASISMICA, dtype: int64

In [371]:
#ETA, numero piani degli edifici scolastici statali 
edifici_o = pd.read_csv("data/MIUR/edilizia/EDIETAORIGINESTA20160831.csv", sep=',',converters={' CODICEEDIFICIO': lambda x: str(x)})
edifici_o = edifici_o.rename(columns=lambda x: x.strip())
edifici_o.info()
edifici_o.head(3)
edifici_o.CODICEEDIFICIO.nunique()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50804 entries, 0 to 50803
Data columns (total 10 columns):
ANNOSCOLASTICO          50804 non-null int64
CODICESCUOLA            50804 non-null object
CODICEEDIFICIO          50804 non-null object
USOSCOLASTICO           50804 non-null object
ALTRIUSI                50804 non-null object
ANNOCOSTRUZIONE         50804 non-null object
FASCIAETACOSTRUZIONE    50804 non-null object
ANNOADATTAMENTO         50804 non-null object
FASCIAETAADATTAMENTO    50804 non-null object
NUMEROPIANI             50804 non-null int64
dtypes: int64(2), object(8)
memory usage: 3.9+ MB


ANNOSCOLASTICO CODICESCUOLA CODICEEDIFICIO USOSCOLASTICO ALTRIUSI ANNOCOSTRUZIONE   FASCIAETACOSTRUZIONE ANNOADATTAMENTO  FASCIAETAADATTAMENTO  NUMEROPIANI
0  201516          AGAA01103A   0840111102     SI            NO       -               tra il 1961 e il 1975  -               Informazione assente  1          
1  201516          AGAA01109L   0840111103     SI            NO       -               tra il 1961 e il 1975  -               Informazione assente  2          
2  201516          AGAA02406G   0840274812     SI            NO       -               dal 1976 in poi        -               Informazione assente  2

36093

In [372]:
#Collegamenti degli edifici scolastici statali: mensa, spazi didattici,... 
edifici_coll = pd.read_csv("data/MIUR/edilizia/EDICOLLEGAMENTISTA20160831.csv", sep=',',converters={' CODICEEDIFICIO': lambda x: str(x)})
edifici_coll = edifici_coll.rename(columns=lambda x: x.strip())
edifici_coll.info()
edifici_coll.head(3)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50804 entries, 0 to 50803
Data columns (total 10 columns):
ANNOSCOLASTICO                  50804 non-null int64
CODICESCUOLA                    50804 non-null object
CODICEEDIFICIO                  50804 non-null object
SCUOLABUS                       50804 non-null object
TRASPORTIPUBBLICIURBANI         50804 non-null object
TRASPORTIPUBBLICIINTERURBANI    50804 non-null object
TRASPORTIFERROVIARI             50804 non-null object
MEZZIPRIVATI                    50804 non-null object
TRASPORTODISABILI               50804 non-null object
ALTRO                           50804 non-null object
dtypes: int64(1), object(9)
memory usage: 3.9+ MB


ANNOSCOLASTICO CODICESCUOLA CODICEEDIFICIO SCUOLABUS TRASPORTIPUBBLICIURBANI TRASPORTIPUBBLICIINTERURBANI TRASPORTIFERROVIARI MEZZIPRIVATI TRASPORTODISABILI           ALTRO
0  201516          AGAA01103A   0840111102     SI        SI                      NO                           SI                  SI           NO                Non Comunicato
1  201516          AGAA01109L   0840111103     SI        SI                      NO                           NO                  SI           SI                Non Comunicato
2  201516          AGAA02406G   0840274812     SI        SI                      NO                           NO                  SI           NO                Non Comunicato

In [373]:
#Ambienti degli edifici scolastici statali: mensa, spazi didattici,... 
edifici_funz = pd.read_csv("data/MIUR/edilizia/EDIAMBFUNZSTA20160831.csv", sep=',',converters={' CODICEEDIFICIO': lambda x: str(x)})
edifici_funz = edifici_funz.rename(columns=lambda x: x.strip())
edifici_funz.info()
edifici_funz.head(3)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50804 entries, 0 to 50803
Data columns (total 8 columns):
ANNOSCOLASTICO         50804 non-null int64
CODICESCUOLA           50804 non-null object
CODICEEDIFICIO         50804 non-null object
SPAZIDIDATTICI         50804 non-null object
AULAMAGNA              50804 non-null object
MENSA                  50804 non-null object
PALESTRAPISCINA        50804 non-null object
SPAZIAMMINISTRATIVI    50804 non-null object
dtypes: int64(1), object(7)
memory usage: 3.1+ MB


ANNOSCOLASTICO CODICESCUOLA CODICEEDIFICIO SPAZIDIDATTICI   AULAMAGNA       MENSA PALESTRAPISCINA SPAZIAMMINISTRATIVI
0  201516          AGAA01103A   0840111102     Esiste         Non Esiste  Non Esiste  Non Esiste      Non Esiste        
1  201516          AGAA01109L   0840111103     Esiste         Non Esiste  Non Esiste  Non Esiste      Non Esiste        
2  201516          AGAA02406G   0840274812     Esiste         Non Esiste  Non Esiste  Non Esiste      Esiste

In [374]:
edifici_funz.MENSA.value_counts()

Non Esiste        35146
Esiste            15117
Non Comunicato    541  
Name: MENSA, dtype: int64

In [375]:
edifici_funz.SPAZIDIDATTICI.value_counts()

Esiste        43603
Non Esiste    7201 
Name: SPAZIDIDATTICI, dtype: int64

In [376]:
edifici_funz.PALESTRAPISCINA.value_counts()

Non Esiste        31552
Esiste            19188
Non Comunicato    64   
Name: PALESTRAPISCINA, dtype: int64

In [377]:
#Dimensioni/Superficie degli edifici scolastici statali 
edifici_d = pd.read_csv("data/MIUR/edilizia/EDICONSISTENZASTA20160831.csv", sep=',',converters={' CODICEEDIFICIO': lambda x: str(x)})
edifici_d = edifici_d.rename(columns=lambda x: x.strip())
edifici_d.info()
edifici_d.head(3)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50804 entries, 0 to 50803
Data columns (total 6 columns):
ANNOSCOLASTICO         50804 non-null int64
CODICESCUOLA           50804 non-null object
CODICEEDIFICIO         50804 non-null object
SPERFICIEAREATOTALE    50804 non-null object
SPERFICIEAREALIBERA    50804 non-null object
VOLUME                 50804 non-null object
dtypes: int64(1), object(5)
memory usage: 2.3+ MB


ANNOSCOLASTICO CODICESCUOLA CODICEEDIFICIO SPERFICIEAREATOTALE SPERFICIEAREALIBERA    VOLUME
0  201516          AGAA01103A   0840111102     3.000,00            2.100,00            3.600,00
1  201516          AGAA01109L   0840111103     18.000,00           6.000,00            9.000,00
2  201516          AGAA02406G   0840274812     1.400,00            455,00              4.930,00

In [378]:
# 1 _ join BASE: Info base SCUOLA + join edifici su CODICESCUOLA
print ("scuole unique: ",scuole_cn.CODICESCUOLA.nunique())
print ("Iniziale: ",len(scuole_cn), "  ", len(edifici))
edifici_1_cn = pd.merge(scuole_cn, edifici, how='left',on='CODICESCUOLA', suffixes=('','_drop'))
edifici_1_cn = edifici_1_cn.drop([col for col in edifici_1_cn.columns if '_drop' in col],axis=1)
print ("Finale: ",len(edifici_1_cn), "-", edifici_1_cn.CODICESCUOLA.nunique() )
edifici_1_cn.to_csv("edifici_1_cn.csv", sep='|',index=False, header=True,  encoding='utf-8')

scuole unique:  786
Iniziale:  786    50804
Finale:  899 - 786


In [379]:
# 2 join, il precedente più i vincoli  - edifici_1 + edifici_v
print ("Iniziale: ", len(edifici_1_cn), "  ", len(edifici_v))
edifici_2_cn = pd.merge(edifici_1_cn, edifici_v, how='left',on=['CODICESCUOLA','CODICEEDIFICIO'], suffixes=('','_drop'))
edifici_2_cn = edifici_2_cn.drop([col for col in edifici_2_cn.columns if '_drop' in col],axis=1)
print ("Finale: ",len(edifici_2_cn),"-",edifici_2_cn.CODICESCUOLA.nunique())
edifici_2_cn.to_csv("edifici_2_cn.csv", sep='|',index=False, header=True,  encoding='utf-8')


Iniziale:  899    50804
Finale:  899 - 786


In [380]:
# 3 join, il precedente più eta - edifici_2 + edifici_o
print ("Iniziale: ", len(edifici_2_cn), "  ", len(edifici_o))
edifici_3_cn = pd.merge(edifici_2_cn, edifici_o, how='left',on=['CODICESCUOLA','CODICEEDIFICIO'],suffixes=('','_drop'))
edifici_3_cn = edifici_3_cn.drop([col for col in edifici_3_cn.columns if '_drop' in col],axis=1)
print ("Finale: ",len(edifici_3_cn),"-",edifici_3_cn.CODICESCUOLA.nunique())
edifici_3_cn.to_csv("edifici_3_cn.csv", sep='|',index=False, header=True,  encoding='utf-8')

Iniziale:  899    50804
Finale:  899 - 786


In [381]:
# 4 join, il precedente più eta - edifici_3 + edifici_funz
print ("Iniziale: ", len(edifici_3_cn), "  ", len(edifici_d))
edifici_4_cn = pd.merge(edifici_3_cn, edifici_d, how='left',on=['CODICESCUOLA','CODICEEDIFICIO'],suffixes=('','_drop'))
edifici_4_cn = edifici_4_cn.drop([col for col in edifici_4_cn.columns if '_drop' in col],axis=1)
print ("Finale: ",len(edifici_4_cn),"-",edifici_4_cn.CODICESCUOLA.nunique())
edifici_4_cn.to_csv("edifici_4_cn.csv", sep='|',index=False, header=True,  encoding='utf-8')

Iniziale:  899    50804
Finale:  899 - 786


In [382]:
# 5 join, il precedente più eta - edifici_3 + edifici_funz
print ("Iniziale: ", len(edifici_4_cn), "  ", len(edifici_funz))
edifici_5_cn = pd.merge(edifici_4_cn, edifici_funz, how='left',on=['CODICESCUOLA','CODICEEDIFICIO'],suffixes=('','_drop'))
print ("Finale: ",len(edifici_5_cn),"-",edifici_5_cn.CODICESCUOLA.nunique())
edifici_5_cn = edifici_5_cn.drop([col for col in edifici_5_cn.columns if '_drop' in col],axis=1)
edifici_5_cn.to_csv("edifici_5_cn.csv", sep='|',index=False, header=True,  encoding='utf-8')

Iniziale:  899    50804
Finale:  899 - 786


In [383]:
# 6 join, il precedente più  edifici_coll
print ("Iniziale: ", len(edifici_5_cn), "  ", len(edifici_coll))
edifici_6_cn = pd.merge(edifici_5_cn, edifici_coll, how='left',on=['CODICESCUOLA','CODICEEDIFICIO'],suffixes=('','_drop'))
print ("Finale: ",len(edifici_6_cn),"-",edifici_5_cn.CODICESCUOLA.nunique())
edifici_6_cn = edifici_6_cn.drop([col for col in edifici_5_cn.columns if '_drop' in col],axis=1)
edifici_6_cn.to_csv("edifici_6_cn.csv", sep='|',index=False, header=True,  encoding='utf-8')

Iniziale:  899    50804
Finale:  899 - 786


In [384]:
# JOIN su EDIFICI E CANTIERI (fondi destinati ai cantieri) - come campo hai solo il codice edificio
print ("Iniziale: ",len(edifici_6_cn), "  ", len(cantieri_cn))
edifici_6_cn_cantieri = pd.merge(edifici_6_cn, cantieri_cn,how='left',on='CODICEEDIFICIO',suffixes=('','_drop'))
#edifici_5_cn_cantieri = edifici_5_cn_cantieri.drop([col for col in edifici_5_cn_cantieri.columns if '_drop' in col],axis=1)
print ("Finale: ",len(edifici_6_cn_cantieri),"-",edifici_6_cn_cantieri.CODICESCUOLA.nunique())
edifici_6_cn_cantieri = edifici_6_cn_cantieri.drop([col for col in edifici_6_cn_cantieri.columns if '_drop' in col],axis=1)

Iniziale:  899    234
Finale:  950 - 786


In [389]:
edifici_6_cn_cantieri_colonne = list(edifici_6_cn_cantieri)
edifici_6_cn_cantieri_colonne


['AREAGEOGRAFICA',
 'CAPSCUOLA',
 'CODICECOMUNESCUOLA',
 'CODICEISTITUTORIFERIMENTO',
 'CODICESCUOLA',
 'DENOMINAZIONEISTITUTORIFERIMENTO',
 'DENOMINAZIONESCUOLA',
 'DESCRIZIONECOMUNE',
 'DESCRIZIONETIPOLOGIAGRADOISTRUZIONESCUOLA',
 'INDIRIZZOSCUOLA',
 'PROVINCIA',
 'REGIONE',
 'CODICEEDIFICIO',
 'SIGLAPROVINCIA',
 'STATOEDIFICIO',
 'ANNOSCOLASTICO',
 'VINCOLIIDROGEOLOGICI',
 'VINCOLIPAESAGGIO',
 'EDIFICIOVETUSTO',
 'VINCOLIZONASISMICA',
 'PROGETTAZIONEANTISISMICA',
 'USOSCOLASTICO',
 'ALTRIUSI',
 'ANNOCOSTRUZIONE',
 'FASCIAETACOSTRUZIONE',
 'ANNOADATTAMENTO',
 'FASCIAETAADATTAMENTO',
 'NUMEROPIANI',
 'SPERFICIEAREATOTALE',
 'SPERFICIEAREALIBERA',
 'VOLUME',
 'SPAZIDIDATTICI',
 'AULAMAGNA',
 'MENSA',
 'PALESTRAPISCINA',
 'SPAZIAMMINISTRATIVI',
 'SCUOLABUS',
 'TRASPORTIPUBBLICIURBANI',
 'TRASPORTIPUBBLICIINTERURBANI',
 'TRASPORTIFERROVIARI',
 'MEZZIPRIVATI',
 'TRASPORTODISABILI',
 'ALTRO',
 'Linea_finanziamento',
 'COMUNE',
 'COD_COMUNE',
 'FONTE',
 'Data_aggiornamento_lavori',
 'Data_i

In [395]:
edifici_6_cn_cantieri.shape

edifici_6_cn_cantieri_column_to_keep = [
 'AREAGEOGRAFICA',
 'CODICECOMUNESCUOLA',
 'CODICEISTITUTORIFERIMENTO',
 'CODICESCUOLA',
 'DENOMINAZIONESCUOLA',
 'DESCRIZIONECOMUNE',
 'DESCRIZIONETIPOLOGIAGRADOISTRUZIONESCUOLA',
 'INDIRIZZOSCUOLA',
 'PROVINCIA',
 'REGIONE',
 'CODICEEDIFICIO',
 'SIGLAPROVINCIA',
 'STATOEDIFICIO',
 'VINCOLIIDROGEOLOGICI',
 'VINCOLIPAESAGGIO',
 'EDIFICIOVETUSTO',
 'VINCOLIZONASISMICA',
 'PROGETTAZIONEANTISISMICA',
 'USOSCOLASTICO',
 'ALTRIUSI',
 'ANNOCOSTRUZIONE',
 'FASCIAETACOSTRUZIONE',
 'ANNOADATTAMENTO',
 'FASCIAETAADATTAMENTO',
 'NUMEROPIANI',
 'SPERFICIEAREATOTALE',
 'SPERFICIEAREALIBERA',
 'VOLUME',
 'SPAZIDIDATTICI',
 'AULAMAGNA',
 'MENSA',
 'PALESTRAPISCINA',
 'SPAZIAMMINISTRATIVI',
 'SCUOLABUS',
 'TRASPORTIPUBBLICIURBANI',
 'TRASPORTIPUBBLICIINTERURBANI',
 'TRASPORTIFERROVIARI',
 'MEZZIPRIVATI',
 'TRASPORTODISABILI',
 'Linea_finanziamento',
 'COMUNE',
 'FONTE',
 'Data_aggiornamento_lavori',
 'Data_inizio',
 'Data_fine',
 'ENTE_RICHIEDENTE',
 'TIPOLOGIA_ENTE_RICHIEDENTE',
 'COD_PRINC',
 'OGGETTO_INTERVENTO',
 'INFO_AGGIUNTIVE',
 'LATITUDINE',
 'LONGITUDINE',
 'TIPO_INTERVENTO',
 'STATO LAVORI',
 'CUP',
 'Importo totale opera',
 'Finanziamento/ Provvedimento']
edifici_6_cn_cantieri= edifici_6_cn_cantieri [edifici_6_cn_cantieri_column_to_keep]
edifici_6_cn_cantieri.shape
edifici_6_cn_cantieri.to_csv("edifici_6_cn_cantieri.csv", sep='|',index=False, header=True,  encoding='utf-8')

(950, 57)

(950, 57)

In [396]:
edifici_6_cn_cantieri.head(5)

AREAGEOGRAFICA CODICECOMUNESCUOLA CODICEISTITUTORIFERIMENTO CODICESCUOLA             DENOMINAZIONESCUOLA DESCRIZIONECOMUNE DESCRIZIONETIPOLOGIAGRADOISTRUZIONESCUOLA         INDIRIZZOSCUOLA PROVINCIA   REGIONE CODICEEDIFICIO SIGLAPROVINCIA STATOEDIFICIO VINCOLIIDROGEOLOGICI VINCOLIPAESAGGIO EDIFICIOVETUSTO VINCOLIZONASISMICA PROGETTAZIONEANTISISMICA USOSCOLASTICO ALTRIUSI ANNOCOSTRUZIONE FASCIAETACOSTRUZIONE ANNOADATTAMENTO FASCIAETAADATTAMENTO  NUMEROPIANI SPERFICIEAREATOTALE SPERFICIEAREALIBERA VOLUME SPAZIDIDATTICI AULAMAGNA MENSA PALESTRAPISCINA SPAZIAMMINISTRATIVI SCUOLABUS TRASPORTIPUBBLICIURBANI TRASPORTIPUBBLICIINTERURBANI TRASPORTIFERROVIARI MEZZIPRIVATI TRASPORTODISABILI Linea_finanziamento COMUNE FONTE Data_aggiornamento_lavori Data_inizio Data_fine ENTE_RICHIEDENTE TIPOLOGIA_ENTE_RICHIEDENTE COD_PRINC OGGETTO_INTERVENTO INFO_AGGIUNTIVE  LATITUDINE  LONGITUDINE TIPO_INTERVENTO STATO LAVORI  CUP Importo totale opera Finanziamento/ Provvedimento
0  NORD OVEST     H727               CNIS014001                CNTD014028   CASA CIRCONDARIALE MORANDI      SALUZZO           IST TEC COMMERCIALE E PER GEOMETRI        REGIONE BRONDA 19/B     CUNEO     PIEMONTE  NaN            NaN            NaN           NaN                  NaN              NaN             NaN                NaN                      NaN           NaN      NaN             NaN                  NaN             NaN                 nan           NaN                 NaN                 NaN    NaN            NaN       NaN   NaN             NaN                 NaN       NaN                     NaN                          NaN                 NaN          NaN               NaN                 NaN    NaN   NaN                       NaN         NaN       NaN              NaN                        NaN       NaN                NaN            nan         nan           NaN             NaN          NaN  NaN                  NaN                        
1  NORD OVEST     G561               CNIC85900A                CNMM85902C   MONDOVI' CORDERO SS PIANFEI     PIANFEI           SCUOLA PRIMO GRADO                        VIA ROMA 99/F           CUNEO     PIEMONTE  NaN            NaN            NaN           NaN                  NaN              NaN             NaN                NaN                      NaN           NaN      NaN             NaN                  NaN             NaN                 nan           NaN                 NaN                 NaN    NaN            NaN       NaN   NaN             NaN                 NaN       NaN                     NaN                          NaN                 NaN          NaN               NaN                 NaN    NaN   NaN                       NaN         NaN       NaN              NaN                        NaN       NaN                NaN            nan         nan           NaN             NaN          NaN  NaN                  NaN                        
2  NORD OVEST     H727               CNIC862006                CNAA862013   SALUZZO-ILARIA ALPI             SALUZZO           SCUOLA INFANZIA                           VIA BARGE, 6/A          CUNEO     PIEMONTE  NaN            NaN            NaN           NaN                  NaN              NaN             NaN                NaN                      NaN           NaN      NaN             NaN                  NaN             NaN                 nan           NaN                 NaN                 NaN    NaN            NaN       NaN   NaN             NaN                 NaN       NaN                     NaN                          NaN                 NaN          NaN               NaN                 NaN    NaN   NaN                       NaN         NaN       NaN              NaN                        NaN       NaN                NaN            nan         nan           NaN             NaN          NaN  NaN                  NaN                        
3  NORD OVEST     H727               CNIC862006                CNAA862024   SALUZZO-VIA MARIA LUISA ALESSI  SALUZZO       

In [397]:
#edifici_5_cn_cantieri.loc[edifici_5_cn_cantieri['CODICESCUOLA']=='CNAA84502A']
edifici_6_cn_cantieri.loc[edifici_6_cn_cantieri['CODICEEDIFICIO']=='0040780011']

AREAGEOGRAFICA CODICECOMUNESCUOLA CODICEISTITUTORIFERIMENTO CODICESCUOLA      DENOMINAZIONESCUOLA DESCRIZIONECOMUNE DESCRIZIONETIPOLOGIAGRADOISTRUZIONESCUOLA             INDIRIZZOSCUOLA PROVINCIA   REGIONE CODICEEDIFICIO SIGLAPROVINCIA STATOEDIFICIO VINCOLIIDROGEOLOGICI VINCOLIPAESAGGIO EDIFICIOVETUSTO             VINCOLIZONASISMICA PROGETTAZIONEANTISISMICA USOSCOLASTICO ALTRIUSI ANNOCOSTRUZIONE  FASCIAETACOSTRUZIONE ANNOADATTAMENTO  FASCIAETAADATTAMENTO  NUMEROPIANI SPERFICIEAREATOTALE SPERFICIEAREALIBERA     VOLUME SPAZIDIDATTICI   AULAMAGNA       MENSA PALESTRAPISCINA SPAZIAMMINISTRATIVI SCUOLABUS TRASPORTIPUBBLICIURBANI TRASPORTIPUBBLICIINTERURBANI TRASPORTIFERROVIARI MEZZIPRIVATI TRASPORTODISABILI                                                                       Linea_finanziamento COMUNE FONTE Data_aggiornamento_lavori Data_inizio Data_fine ENTE_RICHIEDENTE TIPOLOGIA_ENTE_RICHIEDENTE   COD_PRINC OGGETTO_INTERVENTO INFO_AGGIUNTIVE  LATITUDINE  LONGITUDINE             TIPO_INTERVENTO STATO LAVORI  CUP Importo totale opera Finanziamento/ Provvedimento
736  NORD OVEST     D205               CNTF010005                CNTF010005   CUNEO "MARIO DELPOZZO"   CUNEO             ISTITUTO TECNICO INDUSTRIALE              CORSO      A.DE GASPERI 30  CUNEO     PIEMONTE  0040780011     CN             Attivo        NO                   NO               NO              Rischio bassa sismicità (S=6)  NO                       NO            SI       1970            Informazione assente  -               Informazione assente 5.000         22.050,00           14.000,00           18.000,00  Esiste         Non Esiste  Non Esiste  Non Esiste      Esiste              SI        SI                      SI                           NO                  SI           SI                Decreto Mutui BEI seconda annualità - Decreto Interministeriale n. 390 del 6 giugno 2017  Cuneo  MIUR  2017-10-02                NaN         NaN       CUNEO            PROVINCIA                  CNTF010005  NaN                NaN            44.379      7.527         Efficientamento Energetico  Start        NaN  0,0000               700000,0000                
738  NORD OVEST     D205               CNTF010005                CNTF01050E   CUNEO"M.DELPOZZO"SERALE  CUNEO             ISTITUTO TECNICO INDUSTRIALE              CORSO A. DE GASPERI 30      CUNEO     PIEMONTE  0040780011     CN             Attivo        NO                   NO               NO              Rischio bassa sismicità (S=6)  NO                       NO            SI       1970            Informazione assente  -               Informazione assente 5.000         22.050,00           14.000,00           18.000,00  Esiste         Non Esiste  Non Esiste  Non Esiste      Esiste              SI        SI                      SI                           NO                  SI           SI                Decreto Mutui BEI seconda annualità - Decreto Interministeriale n. 390 del 6 giugno 2017  Cuneo  MIUR  2017-10-02                NaN         NaN       CUNEO            PROVINCIA                  CNTF010005  NaN                NaN            44.379      7.527         Efficientamento Energetico  Start        NaN  0,0000               700000,0000

In [398]:
# JOIN tra CANTIERI e OPENCUP
print ("Iniziale: ",len(edifici_6_cn_cantieri), "  ", len(cup))
edifici_6_cn_cantieri_cup = pd.merge(edifici_6_cn_cantieri, cup,how='left',on='CUP',suffixes=('','_drop'))
edifici_6_cn_cantieri_cup = edifici_6_cn_cantieri_cup.drop([col for col in edifici_6_cn_cantieri_cup.columns if '_drop' in col],axis=1)
print ("Finale: ",len(edifici_6_cn_cantieri),"-",edifici_6_cn_cantieri_cup.CODICESCUOLA.nunique())
edifici_6_cn_cantieri_cup.to_csv("edifici_6_cn_cantieri_cup.csv", sep='|',index=False, header=True,  encoding='utf-8')

Iniziale:  950    1129591
Finale:  950 - 786


In [399]:
edifici_6_cn_cantieri_cup.FASCIAETACOSTRUZIONE.value_counts()

Informazione assente     344
tra il 1961 e il 1975    78 
tra il 1921 e il 1945    62 
tra il 1800 e il 1899    57 
dal 1976 in poi          50 
tra il 1946 e il 1960    47 
prima del 1800           42 
tra il 1900 e il 1920    33 
Name: FASCIAETACOSTRUZIONE, dtype: int64

In [400]:
edifici_6_cn_cantieri_cup.FASCIAETAADATTAMENTO.value_counts()

Informazione assente     599
tra il 1946 e il 1975    50 
Dal 1976 in poi          42 
prima del 1945           22 
Name: FASCIAETAADATTAMENTO, dtype: int64

### Numeri di edifici per provincia --> aggregare per provincia

In [150]:
#seziono solo sulle scuole con STATO PROGETTO: ATTIVO
#componenti_strumenti.loc[componenti_strumenti['COR'] == "36527"]  
edifici_attivi = edifici.loc[edifici['STATOEDIFICIO']=="Attivo"]
edifici_attivi.shape

(50804, 11)

### Quanti edifici e scuole ci sono in Italia? 

In [151]:
print ("Numero di scuole: ",edifici_attivi.CODICESCUOLA.nunique())

Numero di scuole:  41753


In [152]:
print ("Numero di edifici: ",edifici.CODICEEDIFICIO.nunique())

Numero di edifici:  36093


### Scuole con il link di scuole sicure

In [141]:
#hanno tutti il link ma con il parametro della scuola solo questo:
len(cup[cup.LINK_SCUOLE_SICURE.str.len() > 58])

9819

In [18]:
cup.AREA_INTERVENTO.value_counts()

IMMOBILI                                 506602
TRASPORTI                                318806
AMBIENTE ED ENERGIA                      266243
RICERCA, ITC, AREE PRODUTTIVE E ALTRO    37870 
Name: AREA_INTERVENTO, dtype: int64

In [115]:
cup.SOTTOSETTORE_INTERVENTO.value_counts()

STRADALI                                                                                          294810
SOCIALI E SCOLASTICHE                                                                             159058
ALTRE INFRASTRUTTURE SOCIALI                                                                      106123
DIFESA DEL SUOLO                                                                                  87641 
RISORSE IDRICHE E ACQUE REFLUE                                                                    79778 
SPORT, SPETTACOLO E TEMPO LIBERO                                                                  59310 
PROTEZIONE, VALORIZZAZIONE E FRUIZIONE DELL'AMBIENTE                                              52504 
BENI CULTURALI                                                                                    52127 
ABITATIVE                                                                                         39867 
DIREZIONALI E AMMINISTRATIVE                           

In [25]:
cup.TIPOLOGIA_INTERVENTO.value_counts()

MANUTENZIONE STRAORDINARIA                                                                     506092
NUOVA REALIZZAZIONE                                                                            231344
MANUTENZIONE ORDINARIA                                                                         113956
RISTRUTTURAZIONE (PER CAMBIO DI DESTINAZIONE D'USO)                                            88677 
ALTRO                                                                                          61613 
AMPLIAMENTO (COMPRESA SOPRAELEVAZIONE) O POTENZIAMENTO                                         40941 
RECUPERO                                                                                       40409 
RESTAURO                                                                                       33515 
MANUTENZIONE STRAORDINARIA CON EFFICIENTAMENTO ENERGETICO                                      4356  
DEMOLIZIONE                                                                       

### Filtro su Tipologie di intervento e Sottosettore

Ci sono 4 **TIPOLOGIE DI INTERVENTO** specifiche:

MANUTENZIONE STRAORDINARIA DI ADEGUAMENTO IMPIANTISTICA E O ANTINCENDIO                        342   
MANUTENZIONE STRAORDINARIA DI MIGLIORAMENTO SISMICO                                            150   
MANUTENZIONE STRAORDINARIA DI ADEGUAMENTO SISMICO                                              132   
MANUTENZIONE STRAORDINARIA PER ACCESSIBILITA' E ABBATTIMENTO DELLE BARRIERE ARCHITETTONICHE    114   

e un **SOTTOSETTORE di INTERVENTO** che ci interessa: SOCIALI E SCOLASTICHE

In [104]:
tipologie = ["MANUTENZIONE STRAORDINARIA DI ADEGUAMENTO IMPIANTISTICA E O ANTINCENDIO", 
             "MANUTENZIONE STRAORDINARIA DI MIGLIORAMENTO SISMICO",
             "MANUTENZIONE STRAORDINARIA DI ADEGUAMENTO SISMICO",
             "MANUTENZIONE STRAORDINARIA PER ACCESSIBILITA' E ABBATTIMENTO DELLE BARRIERE ARCHITETTONICHE"]
settori = ["SOCIALI E SCOLASTICHE"]
print ("CUP TOT:", len(cup))
cup_tipologia = cup[cup['TIPOLOGIA_INTERVENTO'].isin(tipologie)]
print ("CUP TIPOLOGIA ", len(cup_tipologia))

cup_scuole = cup_tipologia[cup_tipologia['SOTTOSETTORE_INTERVENTO'].isin(settori)]
print ("CUP TOT:", len(cup_scuole))
cup_scuole
#cup=cup.loc[cup["CUP"]=="0150700002"]